tracing-core/src/field.rs

```diff
@@ -1197,4 +1211,26 @@ mod test {
         });
         assert_eq!(result, format!("{}", r#"[61 62 63]" "[c0 ff ee]"#));
     }
+
+    #[test]
+    fn record_cow() {
+        let fields = TEST_META_1.fields();
+        let first = Cow::Borrowed(&b"abc"[..]);
+        let second: Cow<'_, [u8]> = Cow::Owned(vec![192, 255, 238]);
+        let values = &[
+            (&fields.field("foo").unwrap(), Some(&first as &dyn Value)),
+            (
+                &fields.field("bar").unwrap(),
+                Some(&Cow::Borrowed(&" ") as &dyn Value),
+            ),
+            (&fields.field("baz").unwrap(), Some(&second as &dyn Value)),
+        ];
+        let valueset = fields.value_set(values);
```

Comment:
nit: I'm a big fan of making tests simple to read and understand at a glance. This might be easier to read formatted as something like:

```rust
let foo = &fields.field("foo").unwrap();
let bar = &fields.field("bar").unwrap();
let baz = &fields.field("bar").unwrap();
let valueset = fields.value_set(&[
    (foo, Some(&first as &dyn Value)),
    (bar, Some(&Cow::Borrowed(&" ") as &dyn Value)),
    (baz, Some(&second as &dyn Value)),
]);
```

Not a big deal / blocker.

Reply:
Unfortunately inlining the `values` list into the `value_set` call doesn't work for lifetime reasons. But I extracted the fields, at least.